pages/class/[id]/edit.vue

```diff
@@ -60,6 +63,44 @@ const onPreviewClassroom = () => {
     router.push(`/class/${editingClassroom.value.id}`);
 };
 
+const showRegistrationToast = (registrationStatus: boolean) => {
+    toast.add({
+        severity: registrationStatus ? "success" : "info",
+        summary: registrationStatus
+            ? "Classroom is open for registration"
+            : "Classroom is closed for registration",
+        group: "tc",
+        life: 1000,
+    });
+};
+
+const updateClassroomStore = (classroom: IClassroom) => {
+    classroomStore.setEditingClassroom(classroom);
+    classroomStore.updateClassroom(classroom);
+};
+
+const onToggleRegistrationStatus = () => {
+    toggleRegistrationStatus(editingClassroom.value.id).then((res) => {
+        const { result: classroom } = res;
+        if (classroom.registrationStatus && !classroom.published) {
+            togglePublishStatus(editingClassroom.value.id).then((res) => {
+                const { result: updatedClassroom } = res;
+                updateClassroomStore(updatedClassroom);
+                if (res.success) {
```

Comment:
Should be update value in store before success?

Reply:
Fixed